packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -973,6 +975,113 @@ function* relevantRemovedRootsFromFields(
 	}
 }
 
+/**
+ * Adds any builds missing from the provided change that are relevant to the change.
+ * Calls {@link relevantRemovedRoots} to determine which builds are relevant.
+ *
+ * @param change - The change with potentially missing builds. Not mutated by this function.
+ * @param getDetachedNode - The function to retrieve a tree chunk from the corresponding detached node id.
+ * @param fieldKinds - The field kinds to delegate to.
+ */
+export function addMissingBuilds(
+	change: TaggedChange<ModularChangeset>,
+	getDetachedNode: (id: DeltaDetachedNodeId) => TreeChunk | undefined,
+	fieldKinds: ReadonlyMap<FieldKindIdentifier, FieldKindWithEditor>,
+): ModularChangeset {
+	const builds: ChangeAtomIdMap<TreeChunk> = new Map<
+		RevisionTag | undefined,
+		Map<ChangesetLocalId, TreeChunk>
+	>();
+
+	for (const root of relevantRemovedRoots(change, fieldKinds)) {
+		const node = getDetachedNode(root);
+
+		// if the detached node could not be found, it should exist in the original builds map
+		if (node === undefined) {
+			assert(change.change.builds !== undefined, "detached node should exist");
+			const original = tryGetFromNestedMap(change.change.builds, root.major, root.minor);
+			assert(original !== undefined, "detached node should exist");
+		}
+
+		setInNestedMap(builds, root.major, root.minor, node);
```

Comment:
Please add a test that detects this

Reply:
Ah, this isn't detectable because the `undefined` entry gets overwritten by `populateNestedMap(change.change.builds, builds, false);`

Reply:
Actually, scratch that, it would get overwritten by `populateNestedMap(change.change.builds, builds, true);` but it won't be right now, so this should be detectable.